chunkset_tpl.h

```diff
@@ -227,17 +227,15 @@ static inline uint8_t* CHUNKMEMSET(uint8_t *out, uint8_t *from, unsigned len) {
 }
 
 Z_INTERNAL uint8_t* CHUNKMEMSET_SAFE(uint8_t *out, uint8_t *from, unsigned len, unsigned left) {
-#if !defined(UNALIGNED64_OK)
-#  if !defined(UNALIGNED_OK)
+#if OPTIMAL_CMP < 32
     static const uint32_t align_mask = 7;
-#  else
+#elif OPTIMAL_CMP == 32
     static const uint32_t align_mask = 3;
-#  endif
 #endif
 
     len = MIN(len, left);
 
-#if !defined(UNALIGNED64_OK)
+#if OPTIMAL_CMP < 64
     while (((uintptr_t)out & align_mask) && (len > 0)) {
         *out++ = *from++;
         --len;
```

Comment:
I kind of speculate this loop here doesn't actually need to exist.  There may be some performance benefit in a niche case for keeping it if say, the ISA has an unaligned write penalty but otherwise permits arbitrary alignment. Then the compiler would likely generate simple writes that happen to be at word aligned offsets.  Of course, there's likely going to be unaligned access in one way or another, either the read, the write, or both.  

Maybe I should see if things are impacted on the G5, which will use the C implementations of these things, supports unaligned writes from GPRs, and I think has a performance penalty for doing so.

Reply:
If we could leave out this loop on some of the architectures, that'd certainly be nice for performance.

Reply:
This was added due to a hang on macOS: https://github.com/zlib-ng/zlib-ng/pull/1031

Reply:
Now that we don't directly access anything unaligned anymore I suspect it's not needed but I don't have a Mac os machine to test it with. Was this with aarch64 or x86?

Reply:
I'm not certain. It looks like it was in a code coverage build.